include/ceed/ceed.h

```diff
@@ -622,6 +619,27 @@ CEED_EXTERN int CeedQFunctionFieldGetSize(CeedQFunctionField qf_field,
 CEED_EXTERN int CeedQFunctionFieldGetEvalMode(CeedQFunctionField qf_field,
     CeedEvalMode *eval_mode);
 
+/** Handle for the user provided callback function to update
+      CeedQFunctionContext fields
+
+ @param[in,out] ctx     CeedQFunctionContext data to update
+ @param[in] field_name  Name of field to update
+ @param[in] data        New value to set for CeedQFunctionContext field
+
+ @return An error code: 0 - success, otherwise - failure
+
+ @ingroup CeedQFunction
+**/
+typedef int (*CeedQFunctionContextUser)(void *ctx, const char *field_name,
```

Comment:
Would a better name for this be something like `CeedQFunctionContextSetter`?  How important is it it have `User` in the name?  I found this kind of confusing at first, like we were making the `QFunctionContext` itself a function, but in reality, the function specifically must be used for setting fields in a context.